pilot/pkg/model/telemetry_test.go

```diff
@@ -669,6 +691,29 @@ func TestTelemetryFilters(t *testing.T) {
 			nil,
 			map[string]string{},
 		},
+		{
+			"disabled-prometheus",
+			[]config.Config{newTelemetry("istio-system", disbaledAllMetrics)},
+			sidecar,
+			networking.ListenerClassSidecarOutbound,
+			networking.ListenerProtocolHTTP,
+			nil,
+			map[string]string{},
+		},
+		{
+			"disabled-then-enabled",
+			[]config.Config{
+				newTelemetry("istio-system", disbaledAllMetrics),
+				newTelemetry("default", emptyPrometheus),
+			},
+			sidecar,
+			networking.ListenerClassSidecarOutbound,
+			networking.ListenerProtocolHTTP,
+			nil,
+			map[string]string{
+				"istio.stats": `{}`,
```

Comment:
Prior to this PR, this would be **disabled** still. Since `disabled` is not set by the `emptyPrometheus`, it uses the parent settings (disabled: true). We need an explicit `disabled: false`

Reply:
what should we do in this scenario:
```golang
newTelemetry("istio-system", emptyPrometheus),
newTelemetry("default", emptyPrometheus),
```

`it uses the parent settings` this sound a little wired.

Reply:
that is how the merge works - unset fields pull from the parent's

Reply:
sorry, I did not get what your concern about this test case. can you share more details?

Reply:
I mean this should be disabled. If you apply this same config in Istio today you get no metrics enabled, but here we enable it

Reply:
test on master branch with the following yaml:
```
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: disable-metrics
  namespace: istio-system
spec:
  metrics:
    - overrides:
        - disabled: true
          match:
            metric: ALL_METRICS
            mode: CLIENT_AND_SERVER
      providers:
        - name: prometheus
---
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: empty-prometheus
  namespace: default
spec:
  metrics:
    - providers:
        - name: prometheus
```
still get metrics
```
# TYPE istio_requests_total counter
istio_requests_total{response_code="200",reporter="destination",source_workload="sleep",source_workload_namespace="default",source_principal="spiffe://cluster.local/ns/default/sa/sleep",source_app="sleep",source_version="unknown",source_cluster="Kubernetes",destination_workload="httpbin",destination_workload_namespace="default",destination_principal="spiffe://cluster.local/ns/default/sa/httpbin",destination_app="httpbin",destination_version="v1",destination_service="httpbin.default.svc.cluster.local",destination_service_name="httpbin",destination_service_namespace="default",destination_cluster="Kubernetes",request_protocol="http",response_flags="-",grpc_response_status="",connection_security_policy="mutual_tls",source_canonical_service="sleep",destination_canonical_service="httpbin",source_canonical_revision="latest",destination_canonical_revision="v1"} 1
```